### DISCUSSION TITLE
User edited distribution list

### DISCUSSION THREAD

**Discussion by gnordli** [postfixadmin/postfixadmin#635] [Ideas]
Hi. 
I have a request where users can add/remove themselves from an distribution list (alias).  

I looked at setting up mailman and others, but it is just too much complexity for what it is worth.  

Any thoughts on something that would be simple to add to postfixadmin?

Geoff

**Comment by DavidGoodwin** [comment]
While at first glance it might appear simple ("just create a PHP script that inserts or deletes from the alias table") I think for every subscribe or unsubscribe you'd have to email a link to the user to ask them if they want to be added/removed. Perhaps you can embed all that stuff in a URL along with some sort of hmac hash to block forgery?

Step 1: signup page choose which list you want to be added on, enter your email address

Step 2: receive email, click link to confirm - that contains e.g. signup.php?email=abc@example.com&action=add&list=1234&hash=....

Step 3: server receives http request at signup.php; validates email+list+action with hash, if valid, adds / removes as necessary.

The main problem I can think of is that someone could use step 1 to send unwanted emails (so captcha? or not make available to the internet?).

**Comment by gnordli** [comment]
Hi @DavidGoodwin   Some thoughts....

1. Only internal users -- not available to the internet.  
2. Existing aliases (with a setting to allow people to add/remove).   Users can't create their own alias.  
3. Maybe a list of users that can be allowed to join the alias -- that is probably the simplest, most secure, because the users are already "approved".  
4. Alias "owner" approval to join -- more dev work, than worth it.  

Potential aliases (ones that allow user add/remove) shows up in their "user" page, that they can join
Aliases could show up in their user page so they can remove them.  

I am trying to solve the issue where someone goes on a holiday and another person takes over their role.   I looked at setting up mailing list, but it is way too much overhead just for this.   

thanks!!

**Reply by gnordli** [reply]
Just thinking about the list of "users" that can be allowed to join.  Ideally, you can include "aliases" that can then expand out to include the users.   That way if you have an "everyone" alias, you can add it, instead of having to type out all of the users.   When you update your everyone alias, it will also update this list.
